bot/exts/utils/reminders.py

```diff
@@ -37,11 +38,43 @@
 LOCK_NAMESPACE = "reminder"
 WHITELISTED_CHANNELS = Guild.reminder_whitelist
 MAXIMUM_REMINDERS = 5
+REMINDER_EDIT_CONFIRMATION_TIMEOUT = 60
 
 Mentionable = discord.Member | discord.Role
 ReminderMention = UnambiguousUser | discord.Role
 
 
+class ModifyReminderConfirmationView(discord.ui.View):
+    """A view to confirm modifying someone else's reminder by admins."""
+
+    def __init__(self, author: discord.Member):
+        super().__init__(timeout=REMINDER_EDIT_CONFIRMATION_TIMEOUT)
+        self.author = author
+        self.result: bool | None = None
+
+    async def interaction_check(self, interaction: Interaction) -> bool:
+        """Only allow interactions from the command invoker."""
+        return interaction.user.id == self.author.id
+
+    async def on_timeout(self) -> None:
+        """Default to not modifying if the user doesn't respond."""
+        self.result = False
+
+    @discord.ui.button(label="Confirm", style=discord.ButtonStyle.blurple, row=0)
+    async def confirm(self, interaction: Interaction, button: discord.ui.Button) -> None:
+        """Confirm the reminder modification."""
+        await interaction.response.edit_message(view=None)
+        self.result = True
+        self.stop()
+
+    @discord.ui.button(label="Cancel", row=0)
+    async def cancel(self, interaction: Interaction, button: discord.ui.Button) -> None:
+        """Cancel the reminder modification."""
+        await interaction.response.edit_message(content="🚫 Operation canceled.", view=None)
```

Comment:
This can now cause ghost pings for the reminder owner. Would sending a second message instead be better?